src/components/confetti_overlay.tsx

```diff
@@ -0,0 +1,92 @@
+import { useEffect, useRef } from "react";
+
+import ConfettiParticle from "@/lib/confetti-particle";
+
+const COLOURS = ["#f50f26", "#e310a7", "#1744e6", "#20e36e", "#e2fa2f"];
+const getRandomColor = () =>
+  COLOURS[Math.floor(Math.random() * COLOURS.length)];
+
+export default function ConfettiCanvas() {
+  const canvasRef = useRef<HTMLCanvasElement>(null);
+  const particles = useRef<ConfettiParticle[]>([]);
+
+  function createParticles() {
+    if (!canvasRef || !canvasRef.current) return;
+
+    const context = canvasRef.current.getContext("2d");
+    particles.current = [];
+
+    let total = 100;
+    if (window.innerWidth > 1080) total = 400;
+    else if (window.innerWidth > 760) total = 300;
+    else if (window.innerWidth > 520) total = 200;
+
+    for (let i = 0; i < total; ++i) {
+      particles.current.push(
+        new ConfettiParticle(
+          context,
+          randBetween(10, 15),
+          randBetween(4, 15),
+          randBetween(0, 360),
+          randBetween(0, window.innerWidth),
+          randBetween(-300, 0),
+          randBetween(2, 3),
+          getRandomColor()
```

Comment:
This is a ton of parameters haha. I think it makes sense for the constructor to hide all these parameters and magic values away. I don't see a use case where we'll need to spawn a particle at an exact position and colour, where the canvas calling function cares.